### PR TITLE
Support alias

### PR DESCRIPTION
**Pull request by mockthebear**

### CODE REVIEW

File: flagset.go [hunk 1]
```diff
@@ -119,6 +119,7 @@ func (f *FlagSetFiller) processField(flagSet *flag.FlagSet, fieldRef interface{}
 		}
 	}
 
+	alias := tag.Get("aliase")
```
Comment (itzg) [inline flagset.go:122]: Originally noting a typo/spelling; however, it got me thinking that one or more aliases would be nice to future proof and only slightly more to implement:

```suggestion
	aliases := tag.Get("aliases")
```

Here, `aliases` will just be a `string` and get split at the end.

File: flagset.go [hunk 2]
```diff
@@ -224,9 +226,12 @@ func (f *FlagSetFiller) processStringToStringMap(fieldRef interface{}, hasDefaul
 		val = *casted
 	}
 	flagSet.Var(&strToStrMapVar{val: val}, renamed, usage)
+	if alias != "" {
+		flagSet.Var(&strToStrMapVar{val: val}, alias, usage)
+	}
```
Comment (itzg) [inline flagset.go:231]: Assuming a switch to plural `aliases`:

```suggestion
	if aliases != "" {
		for _, alias := range strings.Split(aliases, ",") {
			flagSet.Var(&strSliceVar{ref: casted, override: override}, alias, usage)
		}
	}
```

File: flagset.go [hunk 3]
```diff
@@ -276,10 +286,13 @@ func (f *FlagSetFiller) processUint(fieldRef interface{}, hasDefaultTag bool, ta
 		defaultVal = *casted
 	}
 	flagSet.UintVar(casted, renamed, defaultVal, usage)
+	if alias != "" {
+		flagSet.UintVar(casted, alias, defaultVal, usage)
+	}
```
Comment (itzg) [inline flagset.go:291]: ...and same for these

**Review by itzg — COMMENTED** [review]
Looks great...and then I changed the rules 😬 and got to thinking one or more aliases would be great to handle right away 😀.

File: flagset.go (hunk 1 above)
Comment (itzg) [reply flagset.go:122]: ...or maybe it should get split here and let the repetitive code be slightly simpler?

Comment (mockthebear) [reply flagset.go:122]: whooops, indeed it was an typo

File: flagset.go (hunk 2 above)
Comment (mockthebear) [reply flagset.go]: good addition :D
just updated the pr

**Review by itzg — COMMENTED** [review]
Looks great again! Just one more thing: please add a section after this one:

https://github.com/itzg/go-flagsfiller/blob/bc26d8ca734d5f897c67b0e0c8613bba7a2eddf9/docs.go#L40-L47

like:

Aliases

To declares aliases add a tag aliases with a comma separated list of aliases for this flag.

**Comment by mockthebear** [comment]
THere you go :>

File: docs.go [hunk 4]
```diff
@@ -46,6 +46,20 @@ to kebab-case. For example the field named "MyMultiWordField" becomes the flag n
...
+		Timeout time.Duration `default:"1m" aliases:"t"`
+	}
+
+Using it like this you'll can define a custom shorter name. Also possible to separate by commas to
+add more than one alias:
+
+	type Config struct {
+		Timeout time.Duration `default:"1m" aliases:"t,tm,duration"`
+	}
+
```
Comment (itzg) [inline docs.go:62]: Thought I'd tighten up the description. I'll go ahead and commit this to get your PR merged.

```suggestion
Additional aliases, such as short names, can be declared with the `aliases` tag as a comma-separated list:

	type Config struct {
		Timeout time.Duration `aliases:"t"`
		Limit   int `aliases:"l,lim"`
	}
```

**Pull request merged**